ocaml/xapi/xapi_pool_helpers.ml

```diff
@@ -61,7 +61,7 @@ let valid_operations ~__context record _ref' =
     | [_] -> set_errors Api_errors.cluster_already_exists [] [ `cluster_create ]
     (* indicates a bug or a need to update this code (if we ever support multiple clusters in the pool *)
     | _::_ -> failwith "Multiple clusters exist in the pool"
-    | _ -> ()
```

Comment:
`_` is equivalent to "anything matches". The reason the compiler complained is that you already matched every possibility and there was nothing left to match, you should always put the `_` match last:
`[_] ` matches lists with exactly one element.
`_ :: _` matches lists with at least one element (e.g. `one :: []` or `one :: [two; three]`, etc.) but since the previous one already matched  lists with one element, this left only lists of 2 or more to match
`_` matches any kind of list, but due to the previous matches the only remaining list that wasn't matched yet is the list of 0 elements, i.e. `[]`
you now try to match `[]`, but that already got matched by the previous `_`.

So in this case the `_` was equivalent to `[]`, its better to be explicit though and match on `[]` like you did.